kernel/entries.ml

```diff
@@ -87,9 +87,16 @@ type inline = int option (* inlining level, None for no inlining *)
 type parameter_entry = 
     Constr.named_context option * types in_constant_universes_entry * inline
 
+type primitive_entry = {
+  prim_entry_type : types option;
+  prim_entry_univs : Univ.ContextSet.t; (* always monomorphic *)
```

Comment:
If the above comment is right, it is likely that this data is redundant. It should be added globally instead.

Reply:
The documentation of the universe API is not sufficient for me to understand what I should be doing. Can you provide some guidance? I thought I followed @SkySkimmer 's advice when writing this.

Reply:
My point was that in absence of a type annotation, you don't need to register constraints per-definition because you know them statically. Anyways, as said above this doesn't matter if we keep the optional type.